CourseSketchProjects/IdentityServer/src/main/java/coursesketch/database/identity/IdentityManager.java

```diff
@@ -0,0 +1,549 @@
+
+import static database.DbSchoolUtility.getCollectionFromType;
+import static database.DbSchoolUtility.getParentItemType;
+
+/**
+ * Created by dtracers on 10/7/2015.
+ */
+public final class IdentityManager extends AbstractCourseSketchDatabaseReader implements IdentityManagerInterface {
+
+    /**
+     * Declaration and Definition of Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(IdentityManager.class);
+
+    /**
+     * The database that the auth checker grabs data from.
+     */
+    private DB database;
+
+    /**
+     * Creates An IdentityManager with a database.
+     *
+     * @param database
+     *         The database where all of the data is stored.
+     */
+    public IdentityManager(final DB database) {
+        super(null);
+        this.database = database;
+    }
+
+    /**
+     * Creates An Identity Manager with a server information.
+     *
+     * @param serverInfo
+     *         The information about the database location.
+     */
+    public IdentityManager(final ServerInfo serverInfo) {
+        super(serverInfo);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws DatabaseAccessException
+     *         thrown if the database already exist when the database is created.
+     */
+    @Override protected void onStartDatabase() throws DatabaseAccessException {
+        if (this.database != null) {
+            throw new DatabaseAccessException("Mongo instance already exists!");
+        }
+        final MongoClient mongoClient = new MongoClient(super.getServerInfo().getDatabaseUrl());
+        this.database = mongoClient.getDB(super.getServerInfo().getDatabaseName());
+        final DBCollection collection = this.database.getCollection(DatabaseStringConstants.USER_COLLECTION);
+        collection.createIndex(new BasicDBObject(DatabaseStringConstants.USER_NAME, 1));
+        super.setDatabaseStarted();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void createNewItem(final String userId, final String authId, final String itemId, final School.ItemType itemType,
+            final String parentId, final Authenticator authChecker)
+            throws DatabaseAccessException, AuthenticationException {
+        final School.ItemType parentType = getParentItemType(itemType);
+        if (!parentType.equals(itemType)) {
+            final AuthenticationResponder responder = authChecker.checkAuthentication(getParentItemType(itemType), parentId, authId, 0,
+                    Authentication.AuthType.newBuilder().setCheckingAdmin(true).build());
+            if (!responder.hasModeratorPermission()) {
+                throw new AuthenticationException("User does not have permission to insert new items for id: " + parentId,
+                        AuthenticationException.INVALID_PERMISSION);
+            }
+        }
+
+        final BasicDBObject insertQuery = createItemInsertQuery(userId, itemId, itemType);
+        if (!parentType.equals(itemType)) {
```

Comment:
does item type get changed in createItemInsertQuery() function on line 107? If it doesn't just move the body of this if to the previous if cause they are identical